File: webservice_agent/src/main/java/com.abc/kafka/Receiver.java
package com.abc.kafka;

import com.abc.model.rest.DecisionResponse;
import com.google.gson.Gson;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.kafka.annotation.KafkaListener;
import org.springframework.stereotype.Component;

import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;

@Component
public class Receiver {

    private final Logger log = LoggerFactory.getLogger(this.getClass());

    @Autowired
    private Gson gson;

    @Autowired
    private ConcurrentHashMap<UUID, DecisionResponse> decisionMap;

    @KafkaListener(topics = "${webservice.decision}")
    public void receiveDecision(String message) {
        log.info("Receive decision='{}'", message);
        DecisionResponse decisionResponse = gson.fromJson(message, DecisionResponse.class);
        decisionMap.put(decisionResponse.getId(), decisionResponse);
    }

}

File: webservice_agent/src/main/java/com.abc/kafka/Receiver.kt
package com.abc.kafka

import com.abc.model.rest.DecisionResponse
import com.abc.model.rest.UserResponse
import com.google.gson.Gson
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.kafka.annotation.KafkaListener
import org.springframework.stereotype.Component
import java.util.*
import java.util.concurrent.ConcurrentHashMap

@Component
class Receiver {

    @Autowired
    lateinit var gson: Gson

    @Autowired
    lateinit var decisionMap: ConcurrentHashMap<UUID, UserResponse>

    @KafkaListener(topics = arrayOf("decision.response"))
    fun receiveDecision(message: String) {
        println("receiveDecision='$message'")
        val decisionResponse = gson.fromJson(message, DecisionResponse::class.java)
        decisionMap.put(decisionResponse.id, decisionResponse)
    }

}